scopes/component/component-id/component-id.ts

```diff
@@ -159,18 +156,48 @@ export class ComponentID {
     return new ComponentID(legacyId, scope);
   }
 
-  static fromObject(object: any, scope?: string) {
+  // overload when providing scope separaetly, e.g. `fromObject({ name: 'button' }, 'teambit.base-ui')`
+  static fromObject(object: Omit<ComponentIdObj, 'scope'>, scope: string): ComponentID;
+  static fromObject(object: ComponentIdObj, scope?: string): ComponentID;
+  /** deserialize a componnet id from raw object */
+  static fromObject(object: ComponentIdObj, scope?: string) {
     return ComponentID.fromLegacy(new BitId(object), scope);
   }
 
   /**
    * check if object can be correctly deserialized to be a ComponentID
    */
-  static isValidObject(o: any): boolean {
+  static isValidObject(o: any): o is ComponentIdObj {
     return typeof o === 'object' && typeof o.name === 'string' && typeof o.scope === 'string';
     // consider validating values with regex
   }
 
+  static isEqual(a: ComponentID | undefined, b: ComponentID | undefined, opts: EqualityOption = {}): boolean {
+    if (!a && !b) return true;
+    if (!a || !b) return false;
+
+    let result =
+      a.scope === b.scope &&
+      a.toString({ ignoreVersion: opts.ignoreVersion }) === b.toString({ ignoreVersion: opts.ignoreVersion });
```

Comment:
FYI - @GiladShoham @davidfirst.
Made static methods out of the original code:
```ts
const result = id.scope === this.scope && id.toString() === this.toString();
if (!opts.ignoreVersion) {
	return result && this.version === id.version;
}
return result;
```

I passed through `ignoreVersion` to `.toString()`, as it would fail before when versions did not match, even with `ignoreVersion: true`. Maybe we can compare `a.fullName === b.fullName` instead?

Reply:
I guess this is something for @GiladShoham . The `isEqual` function on the ComponentID never worked for me.